agent/internal/containers.go

```diff
@@ -144,32 +145,43 @@ func (c *containerManager) overwriteSpec(
 		slotIds = append(slotIds, strconv.Itoa(d.ID))
 	}
 	spec.RunSpec.ContainerConfig.Labels[dockerContainerDevicesLabel] = strings.Join(slotIds, ",")
-	if len(cont.Devices) > 0 && cont.Devices[0].Type == device.GPU {
-		spec.RunSpec.HostConfig.Runtime = nvidiaRuntime
+
+	spec = c.overwriteSpecforGPUs(cont, spec)
+
+	return spec
+}
+
+func (c *containerManager) overwriteSpecforGPUs(
+	cont cproto.Container, spec container.Spec,
+) container.Spec {
+	useGPU := len(cont.Devices) > 0 && cont.Devices[0].Type == device.GPU
+	spec.RunSpec.ContainerConfig.Env = append(
+		spec.RunSpec.ContainerConfig.Env,
+		fmt.Sprintf("DET_USE_GPU=%t", useGPU),
+	)
+	if useGPU {
+		var uuids []string
```

Comment:
question: I am slightly concerned here that we may lose some ability for GPU/CPU slots on the same agent here. Was this intentional? (it does seem like the logic we had before seemed slightly off)

Assuming that we do want to move forward with this approach, I think it's a best practice to use `uuids := make([]string, 0, len(cont.Devices))` in this case since we are claiming that these are all GPUs. Whereas before we were not confident in the size of the slice.